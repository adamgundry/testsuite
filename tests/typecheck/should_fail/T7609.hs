{-# LANGUAGE TypeOperators #-}

module T7609 where

data X a b

f :: (a `X` a, Maybe)
f = undefined
