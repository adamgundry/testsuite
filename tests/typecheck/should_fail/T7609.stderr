
T7609.hs:7:16:
    Expecting one more argument to `Maybe'
    The second argument of a tuple should have kind `*',
      but `Maybe' has kind `* -> *'
    In the type signature for `f': f :: (a `X` a, Maybe)
